x-pack/plugins/index_management/public/application/sections/enrich_policy_create/steps/configuration.tsx

```diff
@@ -0,0 +1,270 @@
+      json && typeof json === 'object' ? JSON.stringify(json, null, 2) : '{\n\n}',
+    helpText: (
+      <FormattedMessage
+        id="xpack.idxMgmt.enrichPolicyCreate.configurationStep.queryHelpText"
+        defaultMessage="Defaults to: {code} query."
+        values={{
+          code: (
+            <EuiLink external target="_blank" href={documentationService.getMatchAllQueryLink()}>
+              <FormattedMessage
+                id="xpack.idxMgmt.enrichPolicyCreate.configurationStep.matchAllLink"
+                defaultMessage="match_all"
+              />
+            </EuiLink>
+          ),
+        }}
+      />
+    ),
+    validations: [
+      {
+        validator: fieldValidators.isJsonField(
+          i18n.translate('xpack.idxMgmt.enrichPolicyCreate.configurationStep.queryInvalidError', {
+            defaultMessage: 'The query is not valid JSON.',
+          }),
+          { allowEmptyString: true }
+        ),
+      },
+    ],
+  },
+};
+
+export const ConfigurationStep = ({ onNext }: Props) => {
+  const {
+    core: {
+      application: { getUrlForApp },
+    },
+  } = useAppContext();
+  const { draft, updateDraft, updateCompletionState } = useCreatePolicyContext();
+
+  const { form } = useForm({
+    defaultValue: draft,
+    schema: configurationFormSchema,
+    id: 'configurationForm',
+  });
+
+  const onSubmit = async () => {
+    const { isValid, data } = await form.submit();
+
+    if (!isValid) {
+      return;
+    }
+
+    // Update form completion state
+    updateCompletionState((prevCompletionState) => ({
+      ...prevCompletionState,
+      configurationStep: true,
+    }));
+
+    // Update draft state with the data of the form
+    updateDraft((prevDraft: DraftPolicy) => ({
+      ...prevDraft,
+      ...(isEmpty(data.query) ? omit(data, 'query') : data),
+    }));
+
+    // And then navigate to the next step
+    onNext();
+  };
+
+  return (
+    <Form form={form}>
+      <UseField path="name" component={TextField} componentProps={{ fullWidth: false }} />
+
+      <UseField
+        path="type"
+        component={SelectField}
+        componentProps={{
+          fullWidth: false,
+          euiFieldProps: {
+            options: [
+              {
+                value: 'match',
+                text: i18n.translate(
+                  'xpack.idxMgmt.enrichPolicyCreate.configurationStep.matchOption',
+                  { defaultMessage: 'Match' }
+                ),
+              },
+              {
+                value: 'geo_match',
+                text: i18n.translate(
+                  'xpack.idxMgmt.enrichPolicyCreate.configurationStep.geoMatchOption',
+                  { defaultMessage: 'Geo match' }
+                ),
+              },
+              {
+                value: 'range',
+                text: i18n.translate(
+                  'xpack.idxMgmt.enrichPolicyCreate.configurationStep.rangeOption',
+                  { defaultMessage: 'range' }
```

Comment:
nit: this might be better if it starts with a capital letter as the other options.